ares/n64/cpu/cpu.cpp

```diff
@@ -35,6 +35,16 @@ auto CPU::main() -> void {
 
 auto CPU::step(u32 clocks) -> void {
   Thread::clock += clocks;
+
+  clocks >>= 1;
+  i33 dist = scc.compare - scc.count;
+  if(0 <= dist && dist < clocks){
+    scc.cause.interruptPending.bit(Interrupt::Timer) = 1;
+    if constexpr(Accuracy::CPU::Recompiler) {
+      recompiler.earlyExit = true;
+    }
+  }
+  scc.count += clocks;
```

Comment:
An alternative approach would be to store scc.count as an offset relative to Thread::clock. The value of scc.count could be derived from these two values only as needed. This would avoid the need to increment both counters in parallel and would address the rounding issue mentioned in my other comment.

Reply:
The same comment applies: to make the check in the epilogue, we need to know `Thread::clock` at the beginning of each instruction, and I can't see a way to save that in a field in the recompiler.